### PR TITLE
Reusing existing Netty types to speedup http encoding and save scalability issues

### PR DESCRIPTION
**Pull request by franz1981**
This can be completed only with:
- proper benchmarking
- https://github.com/netty/netty/pull/12796 is completed

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java [hunk 1]
```diff
@@ -745,14 +744,14 @@ private Http1xServerResponse write(ByteBuf chunk, PromiseInternal<Void> promise)
         }
       }
       bytesWritten += chunk.readableBytes();
-      HttpObject msg;
       if (!headWritten) {
         prepareHeaders(-1);
-        msg = new AssembledHttpResponse(head, version, status, headers, chunk);
+        conn.writeToChannel(new VertxHttpResponse(head, version, status, headers));
```
Comment (franz1981) [inline src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java:749]: this is the sole case I'm worried about @vietj

**Comment by vietj** [comment]
Netty has been bumped to 4.1.82.Final @franz1981

**Comment by vietj** [comment]
can you update and make a proper PR ?

File: src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java (hunk 1 above)
Comment (vietj) [reply src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java:749]: could Netty have a similar default type ?

Comment (franz1981) [reply src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java:750]: With the `head` element in?

Comment (vietj) [reply src/main/java/io/vertx/core/http/impl/Http1xServerResponse.java:750]: with the headers+chunk

**Comment by franz1981** [comment]
Let me close this, no longer need it

**Pull request closed without merging**